tests/test_basic_read.py

```diff
@@ -0,0 +1,54 @@
+"""Tests for the basic_read function in the csvy library."""
+
+import unittest
+from pathlib import Path
+
+from csvy.readers import basic_read
+
+
+class TestBasicRead(unittest.TestCase):
+    """Tests for the basic_read function in the csvy.readers module."""
+
+    def test_basic_read(self):
+        """Test basic functionality of basic_read with sample CSVY content."""
+        # Sample CSVY content for testing
+        test_file = Path("test_basic_read.csvy")
+
+        # Write sample CSVY content to the test file
+        with open(test_file, "w") as f:
+            f.write(
+                """---
+title: Sample Dataset
+author: User
+description: This is a test CSVY file
+---
+col1, col2, col3
+1, 2, 3
+4, 5, 6
+7, 8, 9
+"""
+            )
```

Comment:
You are already providing the test file as an actual file, so there's no need to populate it dynamically. 
```suggestion
```